### ISSUE TITLE
[BUG] DB goes corrupted after upserting string key of single symbol.

### ISSUE THREAD

**Issue by dvakor** [koculu/ZoneTree#11]
**Describe the bug**

When upserting any value with the string key consist of single symbol
reopening (dispose -> open or app restarting, no matter) failed with exception 

`Write ahead log with segment id = N is corrupted.`

with inner exception

```
System.IO.InvalidDataException: Deserilization of log entry failed. Index=0
 ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Tenray.ZoneTree.Serializers.Utf8StringSerializer.Deserialize(Byte[] bytes)
   at Tenray.ZoneTree.WAL.AsyncCompressedFileSystemWriteAheadLog`2.DeserializeLogEntry(LogEntry& logEntry)
   at Tenray.ZoneTree.WAL.WriteAheadLogEntryReader.ReadLogEntries[TKey,TValue,TLogEntry](ILogger logger, Stream stream, Boolean stopReadOnException, Boolean stopReadOnChecksumFailure, LogEntryRea
derDelegate`1 logEntryReader, LogEntryDeserializerDelegate`3 logEntryDeserializer, Boolean sortByOpIndexes)
   --- End of inner exception stack trace ---
```

Keys with 2 and more symbols works well.

**To Reproduce**

* Create simple console application
* Insert this sample into Program.cs

```
using var db = new ZoneTreeFactory<string, int>()
    .SetDataDirectory("./data")
    .OpenOrCreate();
    
db.Upsert("0", 123);
```

* Run it
* Then try run again, you should get exception

**Expected behavior**

Upserting such keys will not corrupt DB :)

**please complete the following information:**
 - Operating System: Windows 11 x64
 - ZoneTree Version: 1.5.2.0

**Comment by koculu** [comment]
Thank you @dvakor !
Bug fix is included in v1.5.3.
This was a string serializer bug but not a WAL corruption.

https://github.com/koculu/ZoneTree/releases/tag/release-v1.5.3

**Comment by koculu** [comment]
Please use https://github.com/koculu/ZoneTree/releases/tag/release-v1.5.4
as 1.5.3 introduced another bug.
